tests/integration/test_named_collections/test.py

```diff
@@ -34,14 +54,57 @@ def cluster():
         cluster.shutdown()
 
 
-def replace_config(node, old, new):
+def replace_in_server_config(node, old, new):
     node.replace_in_config(
         "/etc/clickhouse-server/config.d/named_collections.xml",
         old,
         new,
     )
 
 
+def replace_in_users_config(node, old, new):
+    node.replace_in_config(
+        "/etc/clickhouse-server/users.d/users.xml",
+        old,
+        new,
+    )
+
+
+def test_access(cluster):
+    node = cluster.instances["node_no_default_access"]
+    assert (
+        "DB::Exception: default: Not enough privileges. To execute this query it's necessary to have grant SHOW NAMED COLLECTIONS ON *.*"
+        in node.query_and_get_error("select count() from system.named_collections")
+    )
+    node = cluster.instances["node_no_default_access_but_with_access_management"]
```

Comment:
can we add test to check access management? I am not fully aware of how access management works, but seems like we check the same thing as for no access management, so thought of checking if we our integration tests can check it.

Reply:
I added this to test this code 
https://github.com/ClickHouse/ClickHouse/blob/690d873126a3b8256f36c708019a0936c06b2d2b/src/Access/UsersConfigAccessStorage.cpp#L236-L240
e.g. if `access_management=1` + without this code I just mentioned it will give access to show named collections, but want to restrict it even for `access_management=1`. By default `access_management=0` so the tests are different.